### ISSUE TITLE
saas-18: Payments from invoice broken.

### ISSUE THREAD

**Issue by nhomar** [odoo/odoo#19558]
Impacted versions: **saas-18**

Steps to reproduce:

Create a sale invoice and try to pay.

Current behavior:

```
Error:
Odoo Server Error

Traceback (most recent call last):
  File "/home/nhomar/odoo/odoo/http.py", line 648, in _handle_exception
    return super(JsonRequest, self)._handle_exception(exception)
  File "/home/nhomar/odoo/odoo/http.py", line 309, in _handle_exception
    raise pycompat.reraise(type(exception), exception, sys.exc_info()[2])
  File "/home/nhomar/odoo/odoo/tools/pycompat.py", line 86, in reraise
    raise value
  File "/home/nhomar/odoo/odoo/http.py", line 685, in dispatch
    result = self._call_function(**self.params)
  File "/home/nhomar/odoo/odoo/http.py", line 341, in _call_function
    return checked_call(self.db, *args, **kwargs)
  File "/home/nhomar/odoo/odoo/service/model.py", line 97, in wrapper
    return f(dbname, *args, **kwargs)
  File "/home/nhomar/odoo/odoo/http.py", line 334, in checked_call
    result = self.endpoint(*a, **kw)
  File "/home/nhomar/odoo/odoo/http.py", line 929, in __call__
    return self.method(*args, **kw)
  File "/home/nhomar/odoo/odoo/http.py", line 514, in response_wrap
    response = f(*args, **kw)
  File "/home/nhomar/odoo/addons/web/controllers/main.py", line 914, in call_button
    action = self._call_kw(model, method, args, {})
  File "/home/nhomar/odoo/addons/web/controllers/main.py", line 902, in _call_kw
    return call_kw(request.env[model], method, args, kwargs)
  File "/home/nhomar/odoo/odoo/api.py", line 689, in call_kw
    return call_kw_multi(method, model, args, kwargs)
  File "/home/nhomar/odoo/odoo/api.py", line 680, in call_kw_multi
    result = method(recs, *args, **kwargs)
  File "/home/nhomar/odoo/addons/account/models/account_payment.py", line 505, in action_validate_invoice_payment
    self.post();
  File "/home/nhomar/enterprise/l10n_mx_edi/models/account_payment.py", line 59, in post
    "invoices that are redone due to payment in advance will need "
  File "/home/nhomar/odoo/addons/account/models/account_payment.py", line 481, in post
    move = rec._create_payment_entry(amount)
  File "/home/nhomar/odoo/addons/account/models/account_payment.py", line 574, in _create_payment_entry
    self.invoice_ids.register_payment(counterpart_aml)
  File "/home/nhomar/odoo/addons/account/models/account_invoice.py", line 898, in register_payment
    return (line_to_reconcile + payment_line).reconcile(writeoff_acc_id, writeoff_journal_id)
  File "/home/nhomar/odoo/addons/account/models/account_move.py", line 993, in reconcile
    remaining_moves = self.auto_reconcile_lines()
  File "/home/nhomar/odoo/addons/account/models/account_move.py", line 959, in auto_reconcile_lines
    'currency_id': currency,
  File "/home/nhomar/odoo/addons/account/models/account_move.py", line 1785, in create
    res.create_tax_cash_basis_entry(percentage_before_rec)
  File "/home/nhomar/odoo/addons/account/models/account_move.py", line 1682, in create_tax_cash_basis_entry
    if move_date > self.company_id.period_lock_date and newly_created_move.date != move_date:
TypeError: unorderable types: str() > bool()
```

Expected behavior:

No traceback.

Video/Screenshot link (optional):

**Comment by nhomar** [comment]
@smetl I think this is your problem ;-)

**Comment by nhomar** [comment]
Complimenting the information, I detected that this is happening just when the blocking dates on accounting settings is not set.

**Comment by odony** [comment]
P3 bug ;-)

**Comment by odony** [comment]
Fixed by 24f238e97c3143e3a655025f1488ba9bf97d68c6 in saas-17, will be forward-ported soon. Thanks for reporting!
